### DISCUSSION TITLE
Allow customization of current scene position in SceneTree

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#6507] [Engine Core]
In one of my projects, every scene is inside a second layer of ViewportContainer & Viewport to achieve retro look 3D scene while keep the UI expandable instead of zoomed.

```
root (SceneTree.root)
    autoloads
    CustomViewportContainer (SceneTree.current_scene)
        CustomViewport
            actual_scene_content
```

`SceneTree.current_scene` is always added as a child of `root`. It would be better if I can change where it is located, e.g. by overriding a virtual method `SceneTree._add_current_scene(node)`.

```
root (SceneTree.root)
    autoloads
    CustomViewportContainer (added in custom SceneTree._init)
        CustomViewport (added in custom SceneTree._init)
            actual_scene_content (SceneTree.current_scene)
```

**Comment by TML233** [comment]
[@godotengine/godot#97914 ](https://github.com/godotengine/godot/pull/97914) can solve this.
